samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -1024,6 +985,108 @@ static int pa_sync_create(void)
 	return bt_le_per_adv_sync_create(&create_params, &pa_sync);
 }
 
+#if !defined(CONFIG_TARGET_BROADCAST_CHANNEL)
+static uint32_t keep_n_least_significant_ones(uint32_t bitfield, uint8_t n)
+{
+	uint32_t result = 0U;
+
+	for (uint8_t i = 0; i < n && bitfield != 0; i++) {
+		uint32_t lsb = bitfield & -bitfield; /* extract lsb */
+
+		result |= lsb;
+		bitfield &= ~lsb; /* clear the extracted bit */
+	}
+
+	return result;
+}
+#endif
+
+static uint8_t get_stream_count(uint32_t bitfield)
+{
+	uint8_t count = 0U;
+
+	for (uint8_t i = 0U; i < BT_ISO_MAX_GROUP_ISO_COUNT; i++) {
+		if ((bitfield & BIT(i)) != 0) {
+			count++;
+		}
+	}
+
+	return count;
+}
+
+static uint32_t select_bis_sync_bitfield(struct base_subgroup_data *base_sg_data,
+					 uint32_t bis_sync_req[CONFIG_BT_BAP_BASS_MAX_SUBGROUPS])
+
+{
+	uint32_t result = 0U;
+
+#if defined(CONFIG_TARGET_BROADCAST_CHANNEL)
+	for (uint8_t i = 0; i < CONFIG_BT_BAP_BASS_MAX_SUBGROUPS; i++) {
+		enum bt_audio_location combine_alloc = BT_AUDIO_LOCATION_MONO_AUDIO;
+		uint32_t combine_bis_sync = 0U;
+
+		if (bis_sync_req[i] == 0) {
+			continue;
+		}
+		/* BIS sync requested in this subgroup. Look for allocation match */
+		for (uint8_t idx = 1; idx <= BT_ISO_BIS_INDEX_MAX; idx++) {
+			struct bis_audio_allocation *base_alloc =
+				&base_sg_data->bis[i].audio_allocation[idx];
```

Comment:
```suggestion
			struct bis_audio_allocation *bis_alloc =
				&base_sg_data->bis[i].audio_allocation[idx];
```

`base_alloc` is misleading if this is the allocation for a specific BIS